app/autosynccontroller.cpp

```diff
@@ -8,33 +8,107 @@
  ***************************************************************************/
 
 #include "autosynccontroller.h"
+#include "coreutils.h"
 
-AutosyncController::AutosyncController( QObject *parent ) : QObject( parent )
-{
-  // raise NotImplemented :)
-}
+#include "qgsproject.h"
+#include "qgsvectorlayer.h"
 
-AutosyncController::~AutosyncController()
+AutosyncController::AutosyncController(
+  LocalProject *openedProject,
+  QgsProject *openedQgsProject,
+  QObject *parent
+)
+  : QObject( parent )
+  , mActiveProject( openedProject )
+  , mActiveQgsProject( openedQgsProject )
 {
-  // raise NotImplemented :)
+  if ( !mActiveQgsProject || !mActiveProject )
+  {
+    CoreUtils::log( QStringLiteral( "Autosync" ), QStringLiteral( "Received an invalid active project data" ) );
+    return;
+  }
+
+  // Register for data change of project's vector layers
+  for ( const QgsMapLayer *layer : mActiveQgsProject->mapLayers( true ) )
```

Comment:
not sure, but isn't better to call `mActiveQgsProject->mapLayers( true )` outside the for loop?

Reply:
best to extract to separate call and store to const var, for readability